roachtest: run mixed version tests on more releases

tbg opened the pull request:
Keep a map of major.minor -> predecessor version so that the version
upgrade tests can run both on the current and previous release(s).

Fixes #34124.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/34143)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34143#-:-LWl6eWU497eAp-eONuL:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/cmd/roachtest/version.go, line 230 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/34143#-LWl6iHo3uTvYu3F40vr:-LWl6iHo3uTvYu3F40vs:b21wnq9) ([raw file](https://github.com/cockroachdb/cockroach/blob/1fca5c74eeedcae7ba3a7732367c5d1fa65d534e/pkg/cmd/roachtest/version.go#L230)):*
> ```Go
> 		r.Add(testSpec{
> 			Name:       fmt.Sprintf("version/mixedWith=%s/nodes=%d", version, n),
> 			MinVersion: "v2.2.0",
> ```

An alternate approach, would be to have `v2.1.x` binaries run mixed width `v2.0.5`, and `v2.2.x` binaries run mixed with `v2.1.3`. `registry.buildVersion` isn't currently exported, but it could be. 


<!-- Sent from Reviewable.io -->

tbg commented:
Good idea, how's this?

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with one small nit.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34143#-:-LWqkry95vW82LCB7NWw:b2tjc6r)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/cmd/roachtest/test.go, line 47 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/34143#-LWqkbslCUjvGIcRJ3Qp:-LWqkbslCUjvGIcRJ3Qq:bpb97lb) ([raw file](https://github.com/cockroachdb/cockroach/blob/7b8a0e71b3a17b48f5258a2e8ed208fd96875b9c/pkg/cmd/roachtest/test.go#L47)):*
> ```Go
> 	postIssues              = true
> 	gceNameRE               = regexp.MustCompile(`^[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?$`)
> 	majorMinorToPredecessor = map[string]string{
> ```

Nit: rather than a global, I'd place this in `PredecessorVersion`, or use a switch statement there. The primary reason is for readability. I don't want to have to consult this map in order to understand what `PredecessorVersion` is doing.


<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: run mixed version tests on more releases

Keep a map of major.minor -> predecessor version so that the version
upgrade tests can run both on the current and previous release(s).

Release note: None
